src/MySqlConnector/Core/CachedProcedure.cs

```diff
@@ -66,13 +67,16 @@ internal sealed class CachedProcedure
 			cmd.Transaction = connection.CurrentTransaction;
 			cmd.CommandText = @"SELECT COUNT(*)
 				FROM information_schema.routines
-				WHERE ROUTINE_SCHEMA = @schema AND ROUTINE_NAME = @component;
+				WHERE ROUTINE_SCHEMA = ? AND ROUTINE_NAME = ?;
 				SELECT ORDINAL_POSITION, PARAMETER_MODE, PARAMETER_NAME, DTD_IDENTIFIER
 				FROM information_schema.parameters
-				WHERE SPECIFIC_SCHEMA = @schema AND SPECIFIC_NAME = @component
+				WHERE SPECIFIC_SCHEMA = ? AND SPECIFIC_NAME = ?
 				ORDER BY ORDINAL_POSITION";
-			cmd.Parameters.AddWithValue("@schema", schema);
-			cmd.Parameters.AddWithValue("@component", component);
+			cmd.Prepare();
+			cmd.Parameters.Add(new() { Value = schema });
+			cmd.Parameters.Add(new() { Value = component });
+			cmd.Parameters.Add(new() { Value = schema });
+			cmd.Parameters.Add(new() { Value = component });
```

Comment:
Are the changes in this file required for this PR, or just an optimisation (to reuse this command server-side) that seemed desirable?

Reply:
Additionally, I think it should be unnecessary to switch from named to positional parameters (and double the parameter count), just because you're calling `Prepare`. The parameter binding code should figure that out IIRC.

Reply:
I think this might actually be a "pessimisation"; since there are two statements, two `COM_EXECUTE_STMT` payloads will have to be sent (instead of one `COM_QUERY`), so there will be multiple DB roundtrips. A quick benchmark bears that out:

|   Method |     Mean |     Error |    StdDev | Allocated |
|--------- |---------:|----------:|----------:|----------:|
|   Normal | 3.190 ms | 0.0975 ms | 0.2766 ms |   3.43 KB |
| Prepared | 3.983 ms | 0.0783 ms | 0.1490 ms |   3.08 KB |

Reply:
this change can really be changed into just "cmd.Prepare();". 
About "pessimisation", this isn't exact: the first stored procedure will indeed have a PREPARE + EXECUTE command that will be slower that a COM_QUERY command, but each time another stored procedure command will be called, only an EXECUTE command will be issued, and this will be faster (especially when using mariadb now that metadata skip is implemented). I hesitate to improve PREPARE + EXECUTE in the same PR, but this will be in another PR : MariaDB permit pipelining PREPARE + EXECUTE command, EXECUTE using statement id 0xffffffff meaning using last prepared statement, this permits to have nearly the same execution time than a COM_QUERY, but next execution being just an EXECUTE command will then be way better.